### PR TITLE
#160609551 Receive verification email on registration

### PR DESCRIPTION
**Pull request by samuelbwambale**
### What does this PR do?

Add feature to receive a verification link via email upon successful signup 

#### Description of Task to be completed?

Currently, when a user creates an account, there is no link sent to their email to verify the account

This task ensures that upon successful signup, an email with verification link is sent to the user’s email.  

#### How should this be manually tested?
- First, create a SendGrid free account [here](https://signup.sendgrid.com/?id=71713987-9f01-4dea-b3d4-8d0bcd9d53ed) and select option of Web API to create your own ``` SENDGRID_API_KEY ```
- Add ``` SENDGRID_API_KEY=the_generated_key ``` in your .env file and save
- Register a user on the endpoint ```POST /api/users``` entering the user data in format below
{
  "user":{
    "username": “Username”,
    "email": “your@email.com”,
    "password": “password1”
  }
} 

- Open your email used for registration, you should find an email from ```targaryen@authorshaven.com```  with the verification link. If it is not seen in the inbox, check in the spam mail.

#### What are the relevant pivotal tracker stories?

- [#160609551](https://www.pivotaltracker.com/story/show/160609551)

#### Screenshots (if appropriate)
screen shot 2018-10-04 at 17 48 27

### CODE REVIEW

**Comment by mendozabree** [comment]
Add sendgrid to requirements.txt

**Comment by mendozabree** [comment]
Add instructions on how to add the sendgrid API key

**Comment by mendozabree** [comment]
Change the email subject, add the name of the service sending the email, in this case, the project.

**Comment by samuelbwambale** [comment]
@mendozabree the SendGrid API key generation step has been added to the PR comment and the relevant files updated as requested. Please review the changes.

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 from rest_framework.permissions import AllowAny, IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.views import APIView
+from sendgrid.helpers.mail import *
 
 from .renderers import UserJSONRenderer
-from .serializers import (
-    LoginSerializer, RegistrationSerializer, UserSerializer
-)
+from .serializers import (LoginSerializer, RegistrationSerializer,
```
Comment (lym) [inline authors/apps/authentication/views.py]: Please undo this change. The previous chunk of code was way more beautiful than your new change.

**Comment by lym** [comment]
@samuelbwambale Please resolve the conflicts too.

**Comment by samuelbwambale** [comment]
@lym fixed the merge conflicts and views.py file updated
